File: DSA_PYTHON/CodeExercises/FindDuplicatesInSortedArray.py

class FindDuplicatesInSortedArray:
    def find_duplicates(self, array):
        i = 0
        while i < len(array) - 1:
            if array[i] == array[i + 1]:
                j = i + 1
                while array[j] == array[i]:
                    j += 1
                print(f"{array[i]} is appearing {j-i} times")
                i = j - 1
            else:
                i += 1

    def main(self):
        array = [3, 6, 8, 8, 10, 12, 15, 15, 15, 15, 20]
        self.find_duplicates(array)
        return ""

File: DSA_CSHARP/CodeExercises/FindDuplicatesInSortedArray.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DSA_CSHARP.CodeExercises;
internal class FindDuplicatesInSortedArray
{
    void FindDulicates(int[] array)
    {
        for (int i = 0; i < array.Length - 1; i++)
        {
            if (array[i] == array[i + 1])
            {
                int j = i + 1;
                while (array[j] == array[i]) j++;
                Console.WriteLine($"{array[i]} is appearing {j - 1} times");
                i = j - 1;
            }
        }
    }
    public string Main()
    {
        var array = new int[] { 3, 6, 8, 8, 10, 12, 15, 15, 15, 15, 20 };
        FindDulicates(array);
        return "";
    }
}
